shub/config.py

```diff
@@ -295,15 +324,28 @@ def get_apikey(self, target, required=True):
 
     def get_image(self, target):
         """Return image for a given target."""
-        if target not in self.images:
-            raise NotFoundException("Could not find image for %s. Please "
-                                    "define it in your scrapinghub.yml."
-                                    "" % target)
-        return self.images[target]
+        target_conf = self.get_target_conf(target, auth_required=False)
+        project, image = target_conf.project_id, target_conf.image
+        if image is None:
+            raise NotFoundException(
+                "Could not find image for project '{}'. Please define it "
+                "in your scrapinghub.yml.".format(target))
+        elif image is False:
+            raise BadParameterException(
+                "Using custom images is disabled for the project '{}'. "
+                "Please enable it in your scrapinghub.yml.".format(target))
+        default_image = SH_IMAGES_REPOSITORY.format(project=project)
+        if image.startswith(SH_IMAGES_REGISTRY) and image != default_image:
+            click.echo(
+                "Found wrong SH repository for project '{}': expected {}.\n  "
+                "Please use aliases `True` or `scrapinghub` to fix it in your "
+                "config.".format(target, default_image), err=True)
```

Comment:
I think it should halt execution, otherwise inattentive user can easily miss it + we agreed that it's a misconfiguration